textdescriptives/components/descriptive_stats.py

```diff
@@ -0,0 +1,80 @@
+"""Calculation of descriptive statistics"""
+from spacy.tokens import Doc
+from spacy.language import Language
+
+import numpy as np
+
+
+@Language.factory("descriptive_stats")
+def create_descriptive_stats_component(nlp: Language, name: str):
+    return DescriptiveStatistics(nlp)
+
+
+class DescriptiveStatistics:
+    def __init__(self, nlp: Language):
+        """Initialise components"""
+        if not Doc.has_extension("token_length"):
+            Doc.set_extension("token_length", getter=self.token_length)
+
+        if not Doc.has_extension("sentence_length"):
+            Doc.set_extension("sentence_length", getter=self.sentence_length)
+
+        if not Doc.has_extension("syllables"):
+            Doc.set_extension("syllables", getter=self.syllables)
+
+        if not Doc.has_extension("counts"):
+            Doc.set_extension("counts", getter=self.counts)
+
+    def __call__(self, doc):
+        """Run the pipeline component"""
+        return doc
+
+    def token_length(self, doc):
+        """Return dict with measures of token length"""
+        token_lengths = [len(token) for token in doc._._filtered_tokens]
+        return {
+            "token_length_mean": np.mean(token_lengths),
+            "token_length_median": np.median(token_lengths),
+            "token_length_std": np.std(token_lengths),
+        }
```

Comment:
Better to have these as individual features? so doc._.token_length_mean, token_length_median ...

Reply:
The one advantage here is that you don't need to calc.  token_lengths three times which I do see as ideal.

Reply:
keeping as is to avoid clutter